File: Python/1040.py
notas = input().split()
n1, n2, n3, n4 = float(notas[0]), float(notas[1]), float(notas[2]), float(notas[3])

media = (n1*2 + n2*3 + n3*4 +n4*1)/10

if media >= 7.0:
    print(f'Media: {media:.1f}')
    print("Aluno aprovado.")
elif media < 5.0:
    print(f'Media: {media:.1f}')
    print("Aluno reprovado.")
else:
    print(f'Media: {media:.1f}')
    print("Aluno em exame.")
    exame = float(input())
    print(f'Nota do exame: {exame}')
    media_final = (media + exame) / 2
    if media_final >= 5:
        print("Aluno aprovado.")
        print(f'Media final: {media_final}')
    else:
        print("Aluno reprovado.")
        print(f'Media final: {media_final}')

File: Java/1040.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        float n1 = sc.nextFloat(), n2 = sc.nextFloat(), n3 = sc.nextFloat(), n4 = sc.nextFloat();
        float media = (n1*2 + n2*3 + n3*4 +n4*1)/10;

        if (media >= 7.0){
            System.out.printf("Media: %.1f\n", media);
            System.out.println("Aluno aprovado.");
        }else if (media < 5.0){
            System.out.printf("Media: %.1f\n", media);
            System.out.println("Aluno reprovado.");
        }else{
            System.out.printf("Media: %.1f\n", media);
            System.out.println("Aluno em exame.");
            float exame = sc.nextFloat();
            System.out.printf("Nota do exame: %.1f\n", exame);
            float media_final = (media + exame) / 2;
            if (media_final >= 5){
                System.out.printf("Aluno aprovado.\n");
                System.out.printf("Media final: %.1f\n", media_final);
            }else{
                System.out.printf("Aluno reprovado.\n");
                System.out.printf("Media final: %.1f\n", media_final);
            }
        }
    }
}
